packages/gatsby-link/src/rewrite-link-path.js

```diff
@@ -34,16 +48,5 @@ export const rewriteLinkPath = (path, relativeTo) => {
     return path
   }
 
-  const { pathname, search, hash } = parsePath(path)
-  const option = getGlobalTrailingSlash()
-  let adjustedPath = path
-
-  if (option === `always` || option === `never`) {
-    const output = applyTrailingSlashOption(pathname, option)
-    adjustedPath = `${output}${search}${hash}`
-  }
-
-  return isAbsolutePath(adjustedPath)
-    ? withPrefix(adjustedPath)
-    : absolutify(adjustedPath, relativeTo)
+  return isAbsolutePath(path) ? applyPrefix(path) : absolutify(path, relativeTo)
```

Comment:
I'll want to double check if we have tests for the `absolutify` portion of this, since behavior here changed. But `absolutify` itself already handles it so I need to remember why it was done this way previously